rset.c

```diff
@@ -301,10 +306,26 @@ int main(int argc, char *argv[])
 					hl_range(host_labels[j]->name, HL_LABEL, 0, 0);
 					printf("\n");
 				}
-				if (tty_opt)
-					(void) ssh_command_tty(hostname, socket_path, host_labels[j], http_port);
-				else
-					(void) ssh_command_pipe(hostname, socket_path, host_labels[j], http_port);
+
+				int exit_code;
+				if (tty_opt) {
+					exit_code = ssh_command_tty(hostname, socket_path, host_labels[j], http_port);
+				} else {
+					exit_code = ssh_command_pipe(hostname, socket_path, host_labels[j], http_port);
+				}
+				if (exit_code != 0) {
+					snprintf(buf, sizeof(buf), "Exited with code %d", exit_code);
+					hl_range(buf, HL_ERROR, 0, 0);
+					printf("\n");
+				}
+				if ((stop_on_err_opt) && exit_code != 0) {
```

Comment:
Let's use a `goto`

```c
if ((stop_on_err_opt) && exit_code != 0)
    goto exit;
```
...

```c
    if (stop_on_err_opt)
        return exit_code;
    else
        return 0;
```

Reply:
Can you verify my logic here in the new commit? I think `exit` needs to be above the `socket_path` cleanup code.